@here/harp-mapview/lib/Utils.ts

```diff
@@ -248,87 +258,114 @@ export namespace MapViewUtils {
     }
 
     /**
-     * The function doing a pan when [[MapView]]'s active [[ProjectionType]] is spherical.
+     * The function doing a pan in the spherical space when [[MapView]]'s active [[ProjectionType]]
+     * is spherical. In other words, the function that rotates the camera around the globe.
      *
      * @param mapView MapView instance.
      * @param fromWorld Start vector representing the scene position of a geolocation.
      * @param toWorld End vector representing the scene position of a geolocation.
      */
-    export function rotateCameraAroundGlobe(
+    export function panCameraAroundGlobe(
         mapView: MapView,
         fromWorld: THREE.Vector3,
         toWorld: THREE.Vector3
     ) {
-        tmpQuaternion.setFromUnitVectors(fromWorld.normalize(), toWorld.normalize()).inverse();
-        tmpMatrix.makeRotationFromQuaternion(tmpQuaternion);
-        mapView.camera.applyMatrix(tmpMatrix);
+        quaternion1.setFromUnitVectors(fromWorld.normalize(), toWorld.normalize()).inverse();
+        matrix1.makeRotationFromQuaternion(quaternion1);
+        mapView.camera.applyMatrix(matrix1);
         mapView.camera.updateMatrixWorld();
     }
 
     /**
-     * Sets the rotation of the camera according to yaw and pitch in degrees.
+     * Sets the rotation of the camera according to yaw and pitch in degrees. The computations hinge
+     * on the current projection and `geoCenter`, because yaw and pitch are defined in tangent
+     * space. In particular, `MapView#geoCenter` needs to be set before calling `setRotation`.
      *
      * **Note:** `yaw == 0 && pitch == 0` will north up the map and you will look downwards onto the
      * map.
      *
      * @param mapView Instance of MapView.
-     * @param yaw Yaw in degrees.
+     * @param yaw Yaw in degrees, counter-clockwise (as opposed to azimuth), starting north.
      * @param pitch Pitch in degrees.
      */
     export function setRotation(mapView: MapView, yaw: number, pitch: number) {
-        yawQuaternion.setFromAxisAngle(yawAxis, geoUtils.MathUtils.degToRad(yaw));
-        pitchQuaternion.setFromAxisAngle(pitchAxis, geoUtils.MathUtils.degToRad(pitch));
-
-        yawQuaternion.multiply(pitchQuaternion);
-        mapView.camera.quaternion.copy(yawQuaternion);
-        mapView.camera.matrixWorldNeedsUpdate = true;
+        const transform = {
+            xAxis: space.x,
+            yAxis: space.y,
+            zAxis: space.z,
+            position: vector
+        };
+        mapView.projection.localTangentSpace(mapView.geoCenter, transform);
+
+        // `matrix2` thereafter is the transform matrix, `matrix1` the rotation matrix.
+        matrix2.makeBasis(transform.xAxis, transform.yAxis, transform.zAxis);
+        matrix2.setPosition(transform.position.x, transform.position.y, transform.position.z);
+        quaternion1.setFromAxisAngle(vector.set(0, 0, 1), THREE.Math.degToRad(yaw));
```

Comment:
`matrix1` can be set up as the multiplication of two simple rotation matrices (no need to create quaternion, multiply, and go back to rotation):

https://en.wikipedia.org/wiki/Rotation_matrix#Basic_rotations